### DISCUSSION TITLE
Would it be possible to customize `toString()`?

### DISCUSSION THREAD

**Discussion by scarf005** [rimbu-org/rimbu#199] [Q&A]
```ts
import { HashMap } from "https​://deno.land/x/rimbu​@1.0.3/hashed/mod.ts"

export type Pos = { y: number; x: number }

const map = HashMap.of<Pos, Pos>([{y:0, x:0}, {y: 0, x: 1}], [{y: 1, x:0}, {y: 1, x: 1}])

console.log(map.toString())
// HashMap([object Object] -> [object Object], [object Object] -> [object Object])
```

it's be nice to be able to pass in custom formatter.

**Comment by vitoke** [comment]
I usually use `map.toArray()` for such purposes, or otherwise `map.stream().map(entry => customToString(entry)).toArray()`.

But I guess it would be possible to add options to `map.toString()` like so: `map.toString({ entryToString: (entry) => customToString(entry) })`. Would that have your preference?

**Comment by vitoke** [comment]
Alternatively, I realized that `Stream` also offers the `join()` method that can be used as follows:
`map.stream().join({ valueToString: (entry) => customToString(entry), sep: ', ' })`
